skupstina_django/scraper/scrape_utils.py

```diff
@@ -60,68 +60,40 @@ def parse_subject_details(url):
         soup = BeautifulSoup(site, 'html.parser')
-        act_text = get_visible_text(soup)
+        act_content = get_visible_text(soup)
         act_title = act_titles[i]
-        acts.append({'act_text': act_text, 'act_url': act_url, 'act_title': act_title})
+        acts.append({'act_content': act_content, 'act_url': act_url, 'act_title': act_title})
     subject_details['acts'] = acts
     return subject_details
 
 
-def write_to_db(acts):
-    for act in acts:
-        # Populate the Act table
-        content = act['act_text']
-        if not Act.objects.filter(content_url=act['act_url']).exists():
-            print('Adding ', act['act_title'])
-            new_act = Act(
-                subject=act['act_title'],
-                content=content,
-                content_url=act['act_url'],
-                type=''
-            )
-            new_act.save()
-
-
 def scrape_everything(url_suffix: str, akti_file: str):
-    all_subjects = []
     periods_url = root_url + url_suffix
     with open('scrapes_completed.txt', 'a', encoding='utf8') as f:
         # Create a new file if not already created
         pass
     with open('scraper/data/' + akti_file, encoding='utf8') as periods_fd:
-        for act_period in list(periods_fd):
+        for act_period in list(periods_fd)[:2]:
             with open('scrapes_completed.txt', 'r+', encoding='utf8') as scrapes_completed:
                 if act_period not in scrapes_completed.read():
-                    parse_complete = False
                     act_period = act_period.strip()
                     print('\nScraping period: ', act_period)
                     url = (periods_url + act_period).replace(' ', '%20')
                     print(url)
+                    period_obj = write_period_to_db(act_period, url)
                     subjects, num_els = parse_subjects_list(url)
-                    all_subjects.extend(subjects)
-                    while not parse_complete:
-                        for subject in subjects:
-                            if 'details' not in subject:
-                                try:
-                                    print('Parsing ', subject['subject_url'])
-                                    subject_details = parse_subject_details(subject['subject_url'])
-                                    subject['details'] = subject_details
-                                    write_to_db(subject['details']['acts'])
-                                    print('Parsed  ', subject['subject_url'], ' **')
-                                    parse_complete = True
-                                except:
-                                    print('Error occurred in parsing ', subject['subject_url'])
-                                    parse_complete = False
+                    for subject in subjects:
+                        parse_complete = False
+                        print('Parsing ', subject['subject_url'])
+                        while not parse_complete:
```

Comment:
Hm… This part seems like a possible endless loop, no? I'd maybe give it some max_retries and decrease it every time you try. Maybe even add a sleep in there, to give it some time before trying once again.

Reply:
Depends on what types of errors are you trying to avoid here.

Reply:
Yup, it's an endless loop to force retry the URL in case of connection errors. It's a good idea to add a sleep timer here because if the scraper stops scraping midway and throws an error, it will take forever to build the DB. Last time I ran it for 12 hours and it was still maybe at 33%!